app/controllers/departments_controller.rb

```diff
@@ -0,0 +1,83 @@
+class DepartmentsController < ApplicationController
+  before_action :set_department, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @departments = policy_scope(Department)
+  end
+
+  def show; end
+
+  def new
+    @department = Department.new
+    @department.build_contact
+    @department.contact.contact_emails.build
+    @department.contact.contact_phones.build
+    authorize @department
+  end
+
+  def edit
+    authorize @department
+  end
+
+  def create
+    @department = Department.new(department_params)
+    authorize @department
+    respond_to do |format|
+      if @department.save
+        format.html do
+          redirect_to @department, notice: t('department_created')
+        end
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def update
+    authorize @department
+    respond_to do |format|
+      if @department.update(department_params)
+        format.html do
+          redirect_to @department, notice: t('department_updated')
+        end
+      else
+        format.html { render :edit }
+      end
+    end
+  end
+
+  def destroy
+    authorize @department
+    @department.destroy
+    respond_to do |format|
+      format.html do
+        redirect_to departments_url, notice: t('department_destroyed')
+      end
+    end
+  end
+
+  private
+
+  def set_department
+    @department = Department.find(params[:id])
+  end
+
+  def department_params
+    email_attrs = [
```

Comment:
you could move this into separate method and which just return the array and then call the method